doc/api-extensions.md

```diff
@@ -1511,3 +1511,10 @@ This adds a new `cloud-init` config key namespace which contains the following k
  - `cloud-init.network-config`
 
  It also adds a new endpoint `/1.0/devices` to devlxd which shows an instance's devices.
+
+## network\_dns\_nat
+This introduces `network.nat` as a config option on network zones (DNS).
+
+It defaults to the current behavior of generating records for all
+instances NICs but if set to `false`, it will instruct LXD to only
+generate records for externally reachable addreses.
```

Comment:
What is an externally reachable address in this context? 
If a NIC has a `routes` or `routes.external` config key, are these considered external?
I was expecting to see some subnet logic here that would exclude addresses that were covered by the network's subnet if NAT was enabled for that network. But if I'm reading this correctly, if the network's IP version has NAT enabled then all leases for that IP version are excluded. Is that right?

Reply:
We don't generate zone entries for `routes.external` addresses at this point.

Reply:
So since all records are for instance NICs directly in the network's subnet, we can use the easy approach of just checking if nat is enabled for a given address family and if it is, exclude all records from the zone.